07-Extract-NDVI-From-Rasters-in-R.Rmd

```diff
@@ -161,104 +166,122 @@ names(avg_NDVI_HARV) <- "meanNDVI"
-between 0-1. The metadata included a `ScaleFactor`. NDVI data are scaled by a 
-factor of 10,000. A scale factor is commonly used in larger datasets. Storing decimal
-places actually consumes more space and creates larger file sizes compared to 
-storing integer values.
-
-```{r scale-data }
 
-#scale data by 10,000
-avg_NDVI_HARV$meanNDVI <- avg_NDVI_HARV$meanNDVI / 10000
-#view output
-head(avg_NDVI_HARV)
+##Create Date Column from Julian Day
+Currently the Julian day is recognized as a character class value.  We'd prefer
+if it was recognized by `R` as one of the data-time classes so that `R` can
+interpret and format it as a date when plotting and in other functions. 
 
-```
+For more information on date-time classes, see the NEON Data Skills lesson 
+[Convert Date & Time Data from Character Class to Date-Time Class (POSIX) in R ]({{ site.baseurl}} /R/Time-Series-Convert-Date-Time-Class-POSIX/).  
 
-#Plot NDVI Using GGPLOT
+We know our data is from 2011, therefore the first day or origin for our Julian
+day count is 01 January 2011.  After setting this as the origin we just need to 
+add the Julian day (as an integer) to the origin date.  Since the origin data
+was originally set as a Date class object, the new `Date` column is also Date 
+class. 
+``` {r convert-jd}
+#set the origin for the julian date (1 Jan 2011)
+origin<-as.Date ("2011-01-01")
 
-We now have a clean `data.frame` with properly scaled NDVI and julian days.
-Let's plot our data.
+#convert "julianDay" from class character to integer
+avg_NDVI_HARV$julianDay <- as.integer(avg_NDVI_HARV$julianDay)
 
-```{r plot-data }
+#create a date column, -1 added because origin is the 1st.  
+avg_NDVI_HARV$Date<- origin + (avg_NDVI_HARV$julianDay-1)
 
-#plot NDVI
-ggplot(avg_NDVI_HARV, aes(julianDay, meanNDVI)) +
-  geom_point(size=4,colour = "blue") + 
-  ggtitle("NDVI for HARV 2011\nLandsat Derived") +
-  xlab("Julian Days") + ylab("Mean NDVI") +
-  theme(text = element_text(size=20))
+#did it work? 
+head(avg_NDVI_HARV$Date)
 
+#What are the classes of the two columns now? 
+class(avg_NDVI_HARV$Date)
+class(avg_NDVI_HARV$julianDay)
 
 ```
 
-> ##Challenges
->
-> 1. Customize your final ggplot output. Change the color of the data points
-> 2. Calculate average NDVI for all rasters in the Landsat_NDVI/SJER/2013/ndvi
-> directory. Then create a plot that renders BOTH sites as a comparison.
+#Scale NDVI Data
+The last thing that we need to do it so scale our data. Valid NDVI values range 
+between 0-1. Our data is from a min of 277 to a max of 9140, how does that work.
+The metadata for this NDVI data have a scale factor: 10,000. A scale factor is
+commonly used in larger datasets to remove decimals. Storing decimal places
+actually consumes more space and creates larger file sizes compared to storing
+integer values.
+
+As we have a small data set of only 13 values we do not need to have the values
+scaled.  
 
+```{r scale-data }
+#de-scale data by 10,000
```

Comment:
HAHA. Thank you for the laugh :) i had a funny image of fishing or something in my mind :)